File: backend/src/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true, trim: true },
  type: { type: String, enum: ['income', 'expense'], required: true },
  color: { type: String, default: '#1890ff' },
  icon: { type: String, default: 'tag' },
  isDefault: { type: Boolean, default: false },
}, { timestamps: true });

module.exports = mongoose.model('Category', categorySchema);

File: mobile/lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String type;
  final String color;
  final String icon;

  Category({
    required this.id,
    required this.name,
    required this.type,
    required this.color,
    required this.icon,
  });

  factory Category.fromJson(Map<String, dynamic> j) => Category(
        id: j['id'] ?? '',
        name: j['name'] ?? '',
        type: j['type'] ?? 'expense',
        color: j['color'] ?? '#1890ff',
        icon: j['icon'] ?? 'tag',
      );
}
